### PR TITLE
[java] Removes deprecated events classes #11819

### PR DESCRIPTION
**Pull request by dev-velo**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

Remove deprecated event classes as described in #11819 
### Motivation and Context

Classes were deprecated in 4.0
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

removes deprecated code

### CODE REVIEW

**Comment by titusfortner** [comment]
Because this is both a major change and is something there isn't a lot of conversation around, we would like to have documentation for the new code and some kind of blog post talking about how to upgrade, etc. That's why we've kept this code around long after it should have been removed. Would you be interested in helping us with that?

**Comment by diemol** [comment]
@RevealOscar do you think you can help us document `EventFiringDecorator` and `WebDriverListener`?

**Comment by RevealOscar** [comment]
wrote a [mini blog](https://docs.google.com/document/d/1tlzOhkuyIy1cmTpKwcXxj8gJ3FqfN32BllPZemNEcTU/edit?usp=sharing) talking about the transition, and how to port over code with examples. I was going to commit it as a md file but figured it would be easier to edit on google docs.Made a copy of the original so modify this file. About to delve into adding javadocs to the code.

**Comment by dev-velo** [comment]
@titusfortner @diemol 
wrote a [mini guide](https://docs.google.com/document/d/1tlzOhkuyIy1cmTpKwcXxj8gJ3FqfN32BllPZemNEcTU/edit?usp=sharing) talking about the transition, and how to port over code with examples. I was going to commit it as a md file but figured it would be easier to edit on google docs since it may need some extra work. Made a copy of the original so feel free to modify this file. About to delve into adding javadocs to the code.

File: java/src/org/openqa/selenium/support/events/WebDriverListener.java [hunk 1]
```diff
@@ -45,200 +48,709 @@ public interface WebDriverListener {
...
 
+  /**
+   * This method will be called after {@link Actions#perform()}  } is called.
+   *
+   * @param driver - decorated WebDriver instance
+   * @param actions - sequence of actions to be performed
+   */
   default void afterPerform(WebDriver driver, Collection<Sequence> actions) {}
 
-  default void beforeResetInputState(WebDriver driver) {}
```
Comment (pujagani) [inline java/src/org/openqa/selenium/support/events/WebDriverListener.java:114]: Just curious, why were these removed from the interface?

File: java/test/org/openqa/selenium/support/events/EventFiringDecoratorTest.java [hunk 2]
```diff
@@ -86,31 +103,200 @@ public void beforeGet(WebDriver driver, String url) {
...
         .isEqualTo(
             String.join(
                 "\n",
-                "beforeAnyCall get",
                 "beforeAnyWebDriverCall get",
                 "beforeGet",
                 "afterGet",
                 "afterAnyWebDriverCall get",
-                "afterAnyCall get"));
+                "beforeAnyWebDriverCall getCurrentUrl",
```
Comment (pujagani) [inline java/test/org/openqa/selenium/support/events/EventFiringDecoratorTest.java:215]: I appreciate adding exhaustive tests. But it might be a good idea to split the PRs out. For example, removal could be one PR and one PR could be adding tests. It will help with cleaner commits.

**Comment by pujagani** [comment]
@RevealOscar Thank you for the blog post. It's great. Would you help us add it as a blog post/tutorial as part of the Selenium website?  https://github.com/SeleniumHQ/seleniumhq.github.io.

**Comment by pujagani** [comment]
@RevealOscar Can you please help resolve the conflicts?

**Comment by dev-velo** [comment]
@pujagani yea I can resolve the conflicts and split the pr into 2. in the process of getting a new machine so havent been able to look at the pr

**Comment by pujagani** [comment]
Thank you @RevealOscar! Let us know if we can help you in anyway.

**Comment by dev-velo** [comment]
@pujagani apologies for the delay but I finally was able to get a new laptop and picked the work back up.

feel free to close this PR as I split this work into separate PR's.
Mini blog pr -> https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1533
Add exhaustive tests -> https://github.com/SeleniumHQ/selenium/pull/13198
Add javadoc -> https://github.com/SeleniumHQ/selenium/pull/13199
Remove deprecated classes -> https://github.com/SeleniumHQ/selenium/pull/13200
